src/paginationCalls/pageDescriptor.ts

```diff
@@ -122,93 +112,49 @@ export class PageDescriptor implements Descriptor {
-    // eslint-disable-next-line @typescript-eslint/no-this-alias
-    const self = this;
+  createIterator(
+    apiCall: GaxCall,
+    request: RequestType,
+    options: CallSettings
+  ): AsyncIterable<{} | undefined> {
     const asyncIterable = {
       [Symbol.asyncIterator]() {
-        const paramPromise: Promise<[
-          RequestType,
-          SimpleCallbackFunction
-        ]> = new Promise(resolve => {
-          self.resolveParams = resolve;
-        });
-        let nextPageRequest: RequestType | null = {};
-        let firstCall = true;
+        let nextPageRequest: RequestType | null | undefined = request;
+        const cache: {}[] = [];
         return {
           async next() {
-            const ongoingCall = new call.OngoingCallPromise();
-            const [request, func] = await paramPromise;
-            if (self.cache.length > 0) {
+            if (cache.length > 0) {
               return Promise.resolve({
                 done: false,
-                value: self.cache.shift(),
+                value: cache.shift(),
               });
             }
-            if (!firstCall && !nextPageRequest) {
-              return Promise.resolve({done: true, value: undefined});
+            if (nextPageRequest) {
+              let result: {} | [ResponseType] | null;
+              [result, nextPageRequest] = (await apiCall(
+                nextPageRequest!,
+                options
+              )) as ResultTuple;
+              cache.push(...(result as ResponseType[]));
             }
-            nextPageRequest = await self.getNextPageRequest(
-              func,
-              firstCall ? request : nextPageRequest!,
-              ongoingCall
-            );
-            firstCall = false;
-            if (self.cache.length === 0) {
-              nextPageRequest = null;
+            if (cache.length === 0) {
               return Promise.resolve({done: true, value: undefined});
             }
-            return Promise.resolve({done: false, value: self.cache.shift()});
+            return Promise.resolve({done: false, value: cache.shift()});
           },
         };
       },
     };
     return asyncIterable;
   }
 
-  async getNextPageRequest(
-    func: SimpleCallbackFunction,
-    request: RequestType,
-    ongoingCall: call.OngoingCallPromise
-  ): Promise<RequestType | null> {
-    ongoingCall.call(func, request);
-    let nextPageRequest = null;
-    const [response] = await ongoingCall.promise;
-    const pageToken = (response as ResponseType)[this.responsePageTokenField];
-    if (pageToken) {
-      nextPageRequest = Object.assign({}, request);
-      nextPageRequest[this.requestPageTokenField] = pageToken;
-    }
-    const responses = (response as ResponseType)[this.resourceField];
-    this.cache.push(...responses);
-    return nextPageRequest;
-  }
-
-  makeCall(request: RequestType, func: GaxCall, settings: CallSettings) {
-    if (settings.pageToken) {
```

Comment:
Curious that if we remove this block, does `setting` includes all pagetoken/pageSize information?